### PR TITLE
Adds support for merging services together

### PR DESCRIPTION
**Pull request by ryan10132**
@nackner for CR!

### CODE REVIEW

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java [hunk 1]
```diff
@@ -60,26 +61,35 @@
     }
 
     public ServiceModel parseServiceClass(Class<?> serviceClass, TypescriptServiceGeneratorConfiguration settings) {
+        return this.parseServiceClass(Lists.newArrayList(serviceClass), settings);
+    }
+
+    public ServiceModel parseServiceClass(List<Class<?>> serviceClasses, TypescriptServiceGeneratorConfiguration settings) {
```
Comment (nackner) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java]: 1. Can we do a serviceClass... method to allow me to pass in two classes without constructing a collection?
2. Does order matter here? (is there a reason for list over Collection or even Iterable?)
3. Not sure what convention we want for naming but it seems odd to pass multiple classes to something that is named parseServiceClass (singular)

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java [hunk 2]
```diff
@@ -60,26 +61,35 @@
...
-        serviceModel.name(serviceClass.getSimpleName());
-
-        Set<Method> serviceMethods = getAllServiceMethods(serviceClass);
-        // find and stores all types that are referenced by this service
-        Set<Type> referencedTypes = Sets.newHashSet();
-        for (Method method : serviceMethods) {
-            referencedTypes.addAll(getTypesFromEndpoint(method, settings));
-        }
-        serviceModel.referencedTypes(referencedTypes);
+        serviceModel.name(serviceClasses.get(0).getSimpleName());
```
Comment (nackner) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java]: This behavior is a bit subtle -- nothing about the signature implies to me that order matters.

File: typescript-service-generator-core/src/test/java/com/palantir/code/ts/generator/utils/TestUtils.java [hunk 3]
```diff
@@ -28,6 +28,14 @@
         String stringGetter(@PathParam("a") String a, @PathParam("b") String b);
     }
 
+    @Path("/testService")
+    public interface TestServiceClassCopy {
```
Comment (nackner) [inline typescript-service-generator-core/src/test/java/com/palantir/code/ts/generator/utils/TestUtils.java]: Is this used?

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java (hunk 1 above)
Comment (ryan10132) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java]: addressed, it's pretty different now

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java (hunk 2 above)
Comment (ryan10132) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java]: done

File: typescript-service-generator-core/src/test/java/com/palantir/code/ts/generator/utils/TestUtils.java (hunk 3 above)
Comment (ryan10132) [inline typescript-service-generator-core/src/test/java/com/palantir/code/ts/generator/utils/TestUtils.java]: removed

**Pull request merged**